### ISSUE TITLE
dict names dont match when Loading the trained best weight

### ISSUE THREAD

**Issue by eedavidwu** [lehaifeng/DASNet#6]
Hi, I noticed that the VGG-16 pretrained weight you used is a bit different from the version I downloaded from the torchvision.
(For example: the VGG in the torchvision, the dict_names are features.0.weight...And you use the conv1.0....)
So would u like to share the pretrained VGG-16 sources you used?

**Comment by louxiantuo** [comment]
There is a VGG-16 model in his readme

**Comment by louxiantuo** [comment]
but,do you have train.txt or val.txt which can run normally

**Comment by eedavidwu** [comment]
> but,do you have train.txt or val.txt which can run normally

Thanks, I ignore that backbone is included.
And I write a txt for the code and add some small changes. (For that the .txt is used for the img and label path)

**Comment by louxiantuo** [comment]
yep，Can you show me what your txt file looks like? My loss function has always been negative.

**Comment by louxiantuo** [comment]
my train.txt like:Model/without_shift/val/A/0000.jpg Model/without_shift/val/B/0000.jpg Model/without_shift/val/OUT/0000.jpg
my val.txt like:Real/subset/train/A/09987.jpg Real/subset/train/B/09987.jpg Real/subset/train/OUT/09987.jpg
but,my print is:
Epoch [0/20] Loss: 83.4675 Mask_Loss_conv5: 28.2261 Mask_Loss_fc: 28.1612 Mask_Loss_embedding: 27.0802
Epoch [0/40] Loss: 19.4641 Mask_Loss_conv5: 5.3363 Mask_Loss_fc: 6.0388 Mask_Loss_embedding: 8.0891
Epoch [0/60] Loss: 115.4509 Mask_Loss_conv5: 37.6562 Mask_Loss_fc: 39.0387 Mask_Loss_embedding: 38.7560
Epoch [0/80] Loss: 12.5370 Mask_Loss_conv5: 7.2014 Mask_Loss_fc: 3.5598 Mask_Loss_embedding: 1.7757
Epoch [0/100] Loss: 49.5341 Mask_Loss_conv5: 18.6511 Mask_Loss_fc: 14.4253 Mask_Loss_embedding: 16.4577
Epoch [0/120] Loss: 23.5248 Mask_Loss_conv5: 7.6976 Mask_Loss_fc: 8.3950 Mask_Loss_embedding: 7.4322
Epoch [0/140] Loss: -4.5434 Mask_Loss_conv5: 2.1200 Mask_Loss_fc: -1.7803 Mask_Loss_embedding: -4.8831
Epoch [0/160] Loss: -37.7697 Mask_Loss_conv5: -12.5636 Mask_Loss_fc: -11.3519 Mask_Loss_embedding: -13.8542
Epoch [0/180] Loss: 78.8943 Mask_Loss_conv5: 24.0203 Mask_Loss_fc: 25.9356 Mask_Loss_embedding: 28.9385
Epoch [0/200] Loss: 36.7203 Mask_Loss_conv5: 14.3210 Mask_Loss_fc: 11.6169 Mask_Loss_embedding: 10.7824
Epoch [0/220] Loss: 34.0041 Mask_Loss_conv5: 8.9688 Mask_Loss_fc: 12.1483 Mask_Loss_embedding: 12.8871
Epoch [0/240] Loss: 15.7435 Mask_Loss_conv5: 3.7773 Mask_Loss_fc: 6.7326 Mask_Loss_embedding: 5.2336
Epoch [0/260] Loss: 51.7126 Mask_Loss_conv5: 20.3426 Mask_Loss_fc: 15.3912 Mask_Loss_embedding: 15.9788
Epoch [0/280] Loss: -88.6766 Mask_Loss_conv5: -28.4506 Mask_Loss_fc: -29.2091 Mask_Loss_embedding: -31.0169

**Comment by eedavidwu** [comment]
> yep，Can you show me what your txt file looks like? My loss function has always been negative.

I write the txt as xx.jpg, xx.jpg,  xx.jpg, and then change the CDD.py to make the image_path suitable.
And I think what u wrote above is right.
Maybe it will converge after some epoches...
And I will check that tomorrow...

**Comment by louxiantuo** [comment]
You have same problem?ok,your pytorch version?

**Comment by louxiantuo** [comment]
Wait，You are separated by a comma?（train.txt）I am using is a space.

**Comment by eedavidwu** [comment]
> Wait，You are separated by a comma?（train.txt）I am using is a space.

I think it has worked in your case or experiment if you extract the img correctly..
And I found that if we load the .pth the author published..The key of the dict name doesn't match the model...
The checkpoint: conv5.0, 5.2 ,5.4
The model: conv5 conv51 conv52.

**Comment by louxiantuo** [comment]
Do you have wechet or qq account？We can communicate with each other.Email:louxiantuo@tju.edu.cn.I just put vgg-16.pth in the specified path. There seems to be no other problems.

**Comment by louxiantuo** [comment]
Epoch [0/2020] Loss: 49.5885 Mask_Loss_conv5: 22.1588 Mask_Loss_fc: 14.7812 Mask_Loss_embedding: 12.6485
Epoch [0/2040] Loss: 50.8110 Mask_Loss_conv5: 23.0210 Mask_Loss_fc: 15.6253 Mask_Loss_embedding: 12.1647
Epoch [0/2060] Loss: 31.8613 Mask_Loss_conv5: 11.7982 Mask_Loss_fc: 8.7114 Mask_Loss_embedding: 11.3517
Epoch [0/2080] Loss: 120.2407 Mask_Loss_conv5: 21.3732 Mask_Loss_fc: 60.2815 Mask_Loss_embedding: 38.5860
Epoch [0/2100] Loss: 91.4055 Mask_Loss_conv5: 34.7435 Mask_Loss_fc: 40.4188 Mask_Loss_embedding: 16.2433
Epoch [0/2120] Loss: 185.7590 Mask_Loss_conv5: 55.9206 Mask_Loss_fc: 101.5306 Mask_Loss_embedding: 28.3079
Epoch [0/2140] Loss: 140.2524 Mask_Loss_conv5: 49.6673 Mask_Loss_fc: 34.4897 Mask_Loss_embedding: 56.0955
Epoch [0/2160] Loss: 72.4813 Mask_Loss_conv5: 30.7614 Mask_Loss_fc: 21.8378 Mask_Loss_embedding: 19.8820
Epoch [0/2180] Loss: 126.9208 Mask_Loss_conv5: 42.3678 Mask_Loss_fc: 24.5625 Mask_Loss_embedding: 59.9905
Epoch [0/2200] Loss: 212.8640 Mask_Loss_conv5: 67.0837 Mask_Loss_fc: 88.9648 Mask_Loss_embedding: 56.8155
Epoch [0/2220] Loss: 57.7524 Mask_Loss_conv5: 22.2561 Mask_Loss_fc: 14.7516 Mask_Loss_embedding: 20.7446
Epoch [0/2240] Loss: 168.4778 Mask_Loss_conv5: 43.2674 Mask_Loss_fc: 93.1779 Mask_Loss_embedding: 32.0325
D:/xunleiDownload/DASNet-master/DASNET/train.py:75: UserWarning: volatile was removed and now has no effect. Use `with torch.no_grad():` instead.
  inputs1,inputs2,targets = Variable(inputs1, volatile=True),Variable(input2,volatile=True) ,Variable(targets)
0.00784313725490196
0.002135313208715162
D:\Software\anaconda3\envs\pytorch_gpu\lib\site-packages\torch\nn\functional.py:2416: UserWarning: nn.functional.upsample is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample is deprecated. Use nn.functional.interpolate instead.")
D:\Software\anaconda3\envs\pytorch_gpu\lib\site-packages\torch\nn\functional.py:2506: UserWarning: Default upsampling behavior when mode=bilinear is changed to align_corners=False since 0.4.0. Please specify align_corners=True if the old behavior is desired. See the documentation of nn.Upsample for details.
  "See the documentation of nn.Upsample for details.".format(mode))
Epoch [1/0] Loss: 38.3325 Mask_Loss_conv5: 9.2234 Mask_Loss_fc: 15.8430 Mask_Loss_embedding: 13.2660
D:/xunleiDownload/DASNet-master/DASNET/train.py:75: UserWarning: volatile was removed and now has no effect. Use `with torch.no_grad():` instead.
  inputs1,inputs2,targets = Variable(inputs1, volatile=True),Variable(input2,volatile=True) ,Variable(targets)

Maybe OK in  Epoch1.

**Comment by eedavidwu** [comment]
> t the loss function is still bouncing aro

The train suiits. But the best weight author gave is not right. (Some layers are not given..)

**Comment by louxiantuo** [comment]
> > t the loss function is still bouncing aro
> 
> The train suiits. But the best weight author gave is not right. (Some layers are not given..)
Emmm，so we should change some weight?The problem is author‘s weight?
Reduce the initial learning rate？

**Comment by eedavidwu** [comment]
Ur email has some sys problems, any other email address is welcome




| |
Haotian Wu
|
|
邮箱：eedavidwu@163.com
|

Signature is customized by Netease Mail Master

On 04/29/2020 11:33, louxiantuo wrote:

t the loss function is still bouncing aro

The train suiits. But the best weight author gave is not right. (Some layers are not given..)
Emmm，so we should change some weight?The problem is author‘s weight?
Reduce the initial learning rate？

**Comment by louxiantuo** [comment]
I send an email from louxiantuo@tju.edu.cn<mailto:louxiantuo@tju.edu.cn> , and you can also send email louxiantuo@outlook.com

发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

发件人: eedavidwu<mailto:notifications@github.com>
发送时间: 2020年4月29日 11:52
收件人: lehaifeng/DASNet<mailto:DASNet@noreply.github.com>
抄送: louxiantuo<mailto:louxiantuo@outlook.com>; Comment<mailto:comment@noreply.github.com>
主题: Re: [lehaifeng/DASNet] dict names dont match when Loading the trained best weight (#6)

Ur email has some sys problems, any other email address is welcome




| |
Haotian Wu
|
|
邮箱：eedavidwu@163.com
|

Signature is customized by Netease Mail Master

On 04/29/2020 11:33, louxiantuo wrote:

t the loss function is still bouncing aro

The train suiits. But the best weight author gave is not right. (Some layers are not given..)
Emmm，so we should change some weight?The problem is author‘s weight?
Reduce the initial learning rate？

―
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub, or unsubscribe.

―
